### PR TITLE
Allow users to filter by gen in /rank

### PR DESCRIPTION
**Pull request by MacChaeger**
This is mostly useful for people wanting to see their general standing but not wanting to see old ladders that they don't play any more.

### CODE REVIEW

File: js/client-chat.js [hunk 1]
```diff
@@ -747,8 +747,13 @@
 				var targets = target.split(',');
 				var formatTargeting = false;
 				var formats = {};
+				var gens = {};
 				for (var i = 1, len = targets.length; i < len; i++) {
-					formats[toId(targets[i])] = 1;
+					if (targets[i].length === 4 && targets[i].indexOf('gen') === 0) {
```
Comment (Zarel) [inline js/client-chat.js]: `.startsWith()`

Comment (Zarel) [inline js/client-chat.js]: Why do people use search functions to test for things in predictable locations? "Does a string start with X" should be O(1), not O(n).........

Comment (MacChaeger) [inline js/client-chat.js]: I'll be honest, I missed that when I was looking through javascript string functions.

Comment (Zarel) [inline js/client-chat.js]: For the record, `string.substr(0, fragment.length) === fragment` is the usual way to test if a string starts with a fragment in basically every programming language ever.

**Pull request merged**